### PR TITLE
Feature/basic movement

### PR DESCRIPTION
**Pull request by tapoton**

### CODE REVIEW

File: Jackal/Jackal Shared/Playing Field/Shared/Game.swift [hunk 1]
```diff
@@ -36,40 +46,113 @@ class Game {
...
+        let boardPositionComponent = BoardPositionComponent()
+        boardPositionComponent.boardPosition = BoardPosition(BoardPosition.Unit(pirateGridPosition.x),
+                                                             BoardPosition.Unit(pirateGridPosition.y))
+        pirate.addComponent(boardPositionComponent)
+        
+        pirates.append(pirate)
+    }
+    
+    
+    var stateMachine: GKStateMachine?
```
Comment (azonov) [inline Jackal/Jackal Shared/Playing Field/Shared/Game.swift]: Property between methods, khm, a u sure?)

File: Jackal/Jackal Shared/Playing Field/Shared/GameStates/StartTurnState.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        
+        let pirate = entity as! PirateEntity
+        self.game.selectedPirate = pirate
+        self.stateMachine?.enter(PirateSelectedState.self)
+    }
+    
+    
+    func entityDeselected(_ entity: GKEntity) {
+        
+        fatalError("Can't be here")
```
Comment (azonov) [inline Jackal/Jackal Shared/Playing Field/Shared/GameStates/StartTurnState.swift]: assertionFailure is more reliable and safe, from my point of view, it's logical error, still we are able to proceed application.

File: Jackal/Jackal Shared/Playing Field/Shared/Level.swift [hunk 3]
```diff
@@ -18,18 +18,58 @@ class Level {
...
         }
+        
+        
+        for x in 0..<Int8(size.width) {
+            for y in 0..<Int8(size.height) {
+                
+                let position = BoardPosition(x, y)
+                guard let centre = graph.node(at: position) else { continue }
+                
+                let connectedPositions = [
```
Comment (azonov) [inline Jackal/Jackal Shared/Playing Field/Shared/Level.swift:45]: How do you see it in the future? Some kind of model will have the save matrix?

**Review by azonov — APPROVED** [review]

File: Jackal/Jackal Shared/Playing Field/Shared/Level.swift (hunk 3 above)
Comment (tapoton) [reply Jackal/Jackal Shared/Playing Field/Shared/Level.swift:45]: yes

**Review by LordPinguin — APPROVED** [review]

**Pull request merged**